docs/architecture/adr012-non-interactive-default-rules-for-zero-padding.md

```diff
@@ -0,0 +1,56 @@
+# ADR 012: Non-Interactive Default Rules for Zero Padding
+
+## Changelog
+
+- 03.01.12: initial draft
+
+## Context
+
+When laying out blobs in the square we create padding to follow the non-interactive default rules. With ADR 009 we decreased padding by aligning the blob in the square to the index of the `MinSquareSize` of the given blob. This is a good improvement but we can do better.
+
+If you align the blob at an index of one you get zero padding as each blob can follow the next. This would make the hash of each square a subtree root. In a square with n shares, you would always get n subtree roots. But having a blob inclusion proof of size n for large blobs is too much and unfeasible.
+Assuming there is a threshold, the number of subtree roots in a proof, where the proof size of a blob is acceptable we can use this threshold to determine the index of the blob in the square. This would give us zero padding for blobs that are smaller than the threshold and non-zero padding for blobs that are larger than the threshold but still smaller than before.
+
+Let's assume a good threshold assumption is that the number of subtree roots in a blob inclusion proof is acceptable if it is smaller than the `MaxSquareSize`. This would mean that the blobs smaller than the square size can use the index of one to get zero padding. Blobs that are larger than `MaxSquareSize` but smaller than `MaxSquareSize * 2` can use the index of two to get a maximum of 1 padding square. Blobs that are larger than `MaxSquareSize * 2` but smaller than `MaxSquareSize * 4` can use the index of 4 to get a maximum of 2 padding squares and so on.
```

Comment:
[typo]

```suggestion
Let's assume a good threshold assumption is that the number of subtree roots in a blob inclusion proof is acceptable if it is smaller than the `MaxSquareSize`. This would mean that the blobs smaller than the square size can use the index of one to get zero padding. Blobs that are larger than `MaxSquareSize` but smaller than `MaxSquareSize * 2` can use the index of two to get a maximum of 1 padding square. Blobs that are larger than `MaxSquareSize * 2` but smaller than `MaxSquareSize * 4` can use the index of 4 to get a maximum of 2 padding shares and so on.
```

[question] if a blob must be aligned to an index of 4, I think the maximum number of padding shares is 3.

```ascii
| B1 | P | P | P | B2 | B2 | B2 | B2 | B2 | B2 | ...

B1: blob one
P: padding
B2: blob two
```

Can you please elaborate on why there is a maximum of 2 padding shares?

Reply:
Yes you a right that was a typo :)